frontend/src/components/GrantAccess.vue

```diff
@@ -130,6 +134,7 @@ function statusSelected(evt: any) {
             maxlength="20"
             placeholder="User's Id"
             v-model="formData.userId"
+            @input="(val) => (formData.userId = formData.userId.toUpperCase())"
```

Comment:
@ianliuwk1019 I raised a question on this with IDIM. However, I'm wondering what happens now with BCeID when we enter a bceid username as uppercase? would the user have problems authenticating?

Reply:
Hi @ArogeG 
Hope I am getting the case right.
Assuming if there is no user in database, and a new **BCEID** user is trying to login to the application (FOM).

Then our **auth_function** lambda found the user from IDP (with **BCEID** username), it creates a new **Fam_User** record in **FAM** with lowercase "**username**" based on [OIDC mapping](https://github.com/bcgov/nr-forests-access-management/wiki/OIDC-Attribute-Mapping), (the lambda function use the entry from IDP - unmodified).

The application **admin** person (for FOM) then login to **FAM frontend** and type the user's username as "**USERNAME**" (or converted by system), then our backend can't find the user in **Fam_User** table (because it is in lowercase "username"), so instead, it creates a **second new user record** as "**USERNAME**" in Fam_User table, and **grant that second user ("USERNAME") with FOM_SUBMITTER** privilege.

Now, the Fam_User table has **two records** for the same user ("**username**" and "**USERNAME**").
So the admin person is granting the wrong record (without knowing).
Then when user tries to login the second time to the application "FOM" then he/she **won't get access** because the 
auth_function is looking up to the "username" user and cannot find the role granting for the user.

Reply:
We had agreed to make the change to uppercase at the database level as well, would this not solve the problem of searching for a new user that tried to login. 

Thanks @ianliuwk1019. My question though is that in the scenario above, if the new user logs in and the username is created as lowercase. We then assign a role to the lower case user. At a second login, if the user tries to login with uppercase would the user be able to login with the appropriate roles? I will wait for @basilv response before scheduling a call to discuss this in detail. The problem here is to avoid duplicates from the different channels in which a user record can be created.

Reply:
- If our system have **search ability through IDIR or BCEID**, we will know exactly the username (or user guid from the IDP) for the user to save into the database and no need to worry about case issue. But probably Conrad knows the feasibility for MVP or post MVP.
- Or, **very unlikely** there are different user from BCEID with "**username**" and "**USERNAME**", so maybe we can assume at the backend when looking up for the user, just to case-insensitive match "username" =="USERNAME" then it will find the record instead of creating a new one.
@ArogeG @basilv

Reply:
Oops, replied above before seeing this additional discussion. Nevertheless, my basic point is that the auth function behavior and API lambda function behavior needs to be consistent with respect to case handling when creating users.